src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -289,11 +290,32 @@ std::optional<edm4eic::MutableCluster> CalorimeterClusterRecoCoG::reconstruct(co
 
       // Solve for eigenvalues.  Corresponds to cluster's 2nd moments (widths)
       Eigen::EigenSolver<Eigen::Matrix2f> es_2D(cov2, false); // set to true for eigenvector calculation
-      Eigen::EigenSolver<Eigen::Matrix3f> es_3D(cov3, false); // set to true for eigenvector calculation
+      Eigen::EigenSolver<Eigen::Matrix3f> es_3D(cov3, true); // set to true for eigenvector calculation
 
       // eigenvalues of symmetric real matrix are always real
       eigenValues_2D = es_2D.eigenvalues();
       eigenValues_3D = es_3D.eigenvalues();
+      //find the eigenvector corresponding to the largest eigenvalue
+      auto eigenvectors= es_3D.eigenvectors();
+      double maxEigenvalue=0;
+      int indexOfMaxEigenvalue=0;
+      for (int i_eigen=0;  i_eigen<3; i_eigen++){
+        auto eigenvalue=eigenValues_3D(i_eigen,0);
+        if(eigenvalue.real()>maxEigenvalue){
+          maxEigenvalue=eigenvalue.real();
+          indexOfMaxEigenvalue=i_eigen;
+        }
+      }
```

Comment:
Use std::max on `es_3D` with compare on the eigenvalues.

Reply:
Hi @wdconinc , I'm trying to get the _index_ of the maximum eigenvalue, not the maximum eigenvalue itself.  Does  std::max return the maximum value or the index of the maximum value?

Reply:
`std::max_element`

Reply:
https://en.cppreference.com/w/cpp/algorithm/max_element

Reply:
@wdconinc 
max_element does not work unless the object you are iterating over is an iterator.  I don't see any way to convert this object, eigenValues_3D, which is type ‘Eigen::Vector3cf’ {aka ‘Eigen::Matrix<std::complex<float>, 3, 1>’} to an iterator type (such as std::vector) without using a similar amount of code to what I used.  So I think this code should remain the way it is.  
Best regards,
Sebouh

Reply:
`eigenValues_3D.begin()` is an iterator. See https://godbolt.org/z/vbzGY3P6r for a working example.

Reply:
The gist is
```cpp
    auto max_eigenvalue_it = std::max_element(
        eigenvalues.begin(),
        eigenvalues.end(),
        [](auto a, auto b) {
            return std::norm(a) < std::norm(b);
        }
    );
    auto max_eigenvector = eigenvectors.col(
        std::distance(
            eigenvalues.begin(),
            max_eigenvalue_it
        )
    );
```

Reply:
thanks.  After a bit of work, I got it to work.